Update ReplaceLogicalNotBooleanWhereWithWherePass to use new pass interface

DrJessop opened the pull request:
Summary: As titled, more efficient now and properly updates the modified bit. Also, this pass was missing tests, so added some for a variety of different cases.

Differential Revision: D86782910

abeakkas reviewed (approved):
With phabricator comments.

DrJessop committed:
Update ReplaceLogicalNotBooleanWhereWithWherePass to use new pass interface (#15755)

Summary:

As titled, more efficient now and properly updates the modified bit. Also, this pass was missing tests, so added some for a variety of different cases.

Reviewed By: abeakkas, hsharma35

Differential Revision: D86782910
